### PR TITLE
Feature/35 : Solve npm and github warnings. Fix all dependencies versions of npm

### PR DESCRIPTION
**Pull request by jrmartin**
This PR takes care of issue #35 and issue #32 

Passing Travis Tests  (tested in branch 'feature/geppetto-client#35') 
https://travis-ci.org/openworm/geppetto-application/builds/553352063

Table showing the changes in dependencies versions and the 'warnings' that were removed. Also code changes.
https://docs.google.com/spreadsheets/d/1PmOEdvFiivKXCsqJUqymkuDQRGEOdY_FrD9ov4JOob8/edit?usp=sharing

Only remaining warnings are due to some dependencies using other old dependencies, even in the newer version.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,108 +8,110 @@
...
+    "pako": "1.0.3",
     "pixi.js": "4.2.3",
-    "plotly.js": "^1.42.5",
+    "plotly.js": "1.42.5",
     "prop-types": "latest",
-    "qunitjs": "^2.1.1",
-    "raw-loader": "^0.5.1",
-    "react": "^16.6.3",
-    "react-dnd": "7.0.2",
+    "qunit": "2.9.2",
```
Comment (tarelli) [inline package.json]: @jrmartin are we still using qunit?

Comment (jrmartin) [reply package.json]: @tarelli There's only one test left from qunit days, but the tests happens in casper too. So we can remove it now.

Comment (tarelli) [reply package.json]: Yeah I'd say let's remove it and the dependency then.

**Comment by tarelli** [comment]
Good article advising against pinning on package.json https://medium.com/the-guild/how-should-you-pin-your-npm-dependencies-and-why-2b8d545c7312

**Comment by filippomc** [comment]
After some read and thinking my opinion is that we shouldn't pin dependencies on geppetto-client, unless there is a library we really know to be problematic. The main reasons to avoid pin are:
- package size increases due to duplicated dependencies which may be compatible
- we are missing potential security updates in the dev process
- pinning does nothing against second-level dependencies.

The source of truth should be package-lock.json, which should be often updated and committed after proper testing. The commits to package-lock should also be separated from other commits in order to allow an easier tracking of what happened.
There are also a bot that is meant to do that dependency automatically, but I'd not consider its use for now.

The package-lock update may be more conservative for application development respect to geppetto-client, especially at the end of the release.

Interesting reads:
- https://renovatebot.com/docs/dependency-pinning/
- https://datree.io/npm-version/
- https://medium.com/the-guild/how-should-you-pin-your-npm-dependencies-and-why-2b8d545c7312

**Comment by jrmartin** [comment]
@tarelli I reintroduce caret on the package.json for all dependencies and committed the pakckage-lock.json. The package-lock.json was automatically created under src/main/webapp, so I committed to geppetto-application 'feature/geppetto-client#35' https://github.com/openworm/geppetto-application/pull/15

Travis is passing https://travis-ci.org/openworm/geppetto-application/builds/565590677

**Comment by tarelli** [comment]
@tarelli @filippomc test this locally

**Comment by tarelli** [comment]
@jrmartin can you merge the latest of development here so we can see the npm build run in travis? thx!

**Comment by jrmartin** [comment]
This PR goes in hand with this one from geppetto-application, https://github.com/openworm/geppetto-application/pull/15, both have to be merged to development at same time to avoid errors.

**Comment by afonsobspinto** [comment]
Fixes #13 as well :+1:

**Pull request merged**